### ISSUE TITLE
Warning "Event object wasn't available in remote cache" when using remote store

### ISSUE THREAD

**Issue by ahus1** [keycloak/keycloak#25322]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

storage

### Describe the bug

There are periodic warnings when using Keycloak with the remote store in a Cross-DC environment. 

> {"timestamp":"2023-12-06T09:19:01.678Z","sequence":9974,"loggerClassName":"org.jboss.logging.Logger","loggerName":"org.keycloak.cluster.infinispan.InfinispanNotificationsManager","level":"WARN","message":"Event object wasn't available in remote cache after event was received. Event key: task::ClearExpiredEvents","threadName":"kc-work-cache-event-listener-1273-0","threadId":95,"mdc":{},"ndc":"","hostName":"keycloak-0","processName":"QuarkusEntryPoint","processId":1}

### Version

main

### Expected behavior

The warning shouldn't be logged.

### Actual behavior

There is a warning.

### How to Reproduce?

Run Keycloak with remote Infinispan as described for the multi-site setup. Wait for the regular tasks to be scheduled.

### Anything else?

I'll create a PR for this.

**Comment by xgp** [comment]
Hi @ahus1 Looking at your comment
```
            // Items with the TASK_KEY_PREFIX might be gone fast once the locking is complete, therefore, don't log them.
            // It is still good to have the warning in case of real events return null because they have been, for example, expired
```
Can you elaborate on the "in case of real events return null because they have been, for example, expired"?

Does that mean there should be a warning when a real event occurs, but the entry is expired from the remote cache in the small time it takes for the `InfinispanNotificationsManager` code to execute the listener?

Or does it mean that the `InfinispanNotificationsManager` will receive an event (e.g. a modified event?) *when* an entry is expired?

In a working cluster with external infinispan, I see this warning all the time (version after your PR was merged), and they seem to come in pairs (in a 3 pod cluster). I'm wondering is this an expected behavior, and should this be a different log level than `WARN`?

**Comment by ahus1** [comment]
> Can you elaborate on the "in case of real events return null because they have been, for example, expired"?

The `eventReceived` is only triggered when an item is modified or created. When looking for where it is being called, there is a different branch for removed entries. 

The behavior you see is not expected, so I'd say please raise a new issue for that. To analyze this further, it would be good to know the cache name where this is happening, and possibly more about the element that is being handled here.  If you could enable tracing for that class, the line `logger.tracef("Received event: %s", event);` might revel more information. Thanks!

**Comment by xgp** [comment]
Thanks @ahus1 . I updated logging and built a custom image. I'll report back what I find.

**Comment by fredlb** [comment]
@xgp Did you find anything? I am seeing the same warnings in a similar setup to yours.

**Comment by xgp** [comment]
@fredlb I have added some logging and I'm still trying to understand 1. where the entries are originating, and 2. why they are missing when it goes to process them.

**Comment by fredlb** [comment]
Turning on `TRACE` logging does not reveal where it happens it seems, at most this is what I get out of it:
```Sending event with key 28a58bf7-a6c4-41e6-9e7e-64da80940b09: ClientUpdatedEvent [ realmId=realm-id, clientUuid=8aaa9a2c-ded5-47c4-9ed8-be9ee9640b82, clientId=client-id ]```

I've also observed it happening for `RealmUpdatedEvent`.

**Comment by ahus1** [comment]
I've been able to track down a situation in #28303 how this happens. See the discussion over there. A fix will be available in the next KC24 patch release.

**Comment by tania2** [comment]
This is issue happens in version 26.2.5. Please advise

**Comment by ahus1** [comment]
@tania2 - please create a new bug issue and provide the full log statement. Thanks!
